lib/mongoid/criteria/queryable/extensions/boolean.rb

```diff
@@ -20,7 +20,11 @@ module ClassMethods
             # @return [ true, false ] The boolean value.
             def evolve(object)
               __evolve__(object) do |obj|
-                obj.to_s =~ (/\A(true|t|yes|y|on|1|1.0)\z/i) ? true : false
+                if obj.to_s =~ (/\A(true|t|yes|y|on|1|1.0)\z/i)
+                  true
+                elsif obj.to_s =~ (/\A(false|f|no|n|off|0|0.0)\z/i)
+                  false
```

Comment:
This is a behavior change, before everything defaulted to false, now it defaults to nil.